### PR TITLE
DQN updates

### PR DESCRIPTION
**Pull request by GyuH13**
### 1. JSON File Bug Fix (https://github.com/ROBOTIS-GIT/turtlebot3_machine_learning/pull/87)
- **Issue**: The `step` parameter in JSON file was incorrectly named
- **Solution**: Changed `step` → `step_counter`

### 2. System Arguments Changed to ROS Parameters
- **Before**: Passed as command-line arguments (`ros2 run turtlebot3_dqn dqn_agent 1 1000`)
- **After**: Passed as ROS parameters (`ros2 run turtlebot3_dqn dqn_agent --ros-args -p epsilon_decay:=6000 -p max_training_episodes:=1000`)

### 3. GPU Usage Control Parameter Added
- **New Parameter**: `use_gpu`
- **Default Value**: `false` (dqn_agent), `false` (dqn_test)
- **Description**: Allows selection of whether to use GPU

### 4. Model File Loading Feature Improvement
- **New Parameter**: `model_file`
- **Default Value**: `''` (empty string)
- **Description**: Enables loading existing trained models and continuing training
- **Variable Name Change**: `load_model` → `use_pretrained_model`
- **Model Path**: Changed from class variable to local variable

### 5. Model Filename Format Change
- **Before**: `stage1_episode600.h5`, `stage1_episode600.json`
- **After**: `model1.h5`, `model1.json` (automatically increments number)
- **Impact**: When saving models, automatically uses next number if file exists

### 6. Verbose Parameter Added (https://github.com/ROBOTIS-GIT/turtlebot3_machine_learning/pull/77)
- **New Parameter**: `verbose`
- **Default Value**: `true`
- **Description**: Controls output of training/test process
- **Applied**: Applied to all `model.predict()` calls

### 7. Introduce Lazy import tensorflow and keras for CI and ROSDISTRO distribution
- Implemented lazy import for TensorFlow and Keras modules to prevent immediate loading during package initialization.
- Remove dependency of python-tensorflow-pip and python-numpy for rolling distribution

### 8. Other Changes
- **CHANGELOG Update**: Added version 1.0.2 changes
- **Package Version Update**: 1.0.1 → 1.0.2
- **Author Added**: Hyungyu Kim added
- **README Update**: 
  - Removed `noetic` from Active Branches (moved to Legacy)
  - Added `coin_d4_driver` link

## Changed Files List

1. `README.md` - Branch information and link updates
2. `turtlebot3_dqn/CHANGELOG.rst` - Added version 1.0.2 changes
3. `turtlebot3_dqn/package.xml` - Version and author updates
4. `turtlebot3_dqn/setup.py` - Version and author updates
5. `turtlebot3_dqn/turtlebot3_dqn/dqn_agent.py` - Major logic changes
6. `turtlebot3_dqn/turtlebot3_dqn/dqn_test.py` - Parameter method changes
7. `turtlebot3_dqn/saved_model/model1.h5` - New model file added
8. `turtlebot3_dqn/saved_model/model1.json` - New JSON file added (uses step_counter)
9. `turtlebot3_dqn/saved_model/stage1_episode600.h5` - Removed
10. `turtlebot3_dqn/saved_model/stage1_episode600.json` - Removed (used step)
11. `turtlebot3_machine_learning/CHANGELOG.rst` - Added version 1.0.2 changes
12. `turtlebot3_machine_learning/package.xml` - Version update

### Running dqn_agent
```bash
$ ros2 run turtlebot3_dqn dqn_agent --ros-args \
  -p epsilon_decay:=6000 \
  -p max_training_episodes:=1000 \
  -p use_gpu:=true \
  -p model_file:=model1.h5 \
  -p verbose:=true
```

### Running dqn_test
```bash
$ ros2 run turtlebot3_dqn dqn_test --ros-args \
  -p model_file:=model1.h5 \
  -p use_gpu:=true \
  -p verbose:=true
```

## Notes

- This PR's main purpose is to fix the JSON file label error.
- Model files in the old `stage1_episode600.h5` format are no longer used.
- New model files are automatically saved in `model1.h5`, `model2.h5` format.
- JSON files include `step_counter` and `trained_episodes` information.

### CODE REVIEW

**Review by yun-goon — APPROVED** [review]
Good!

**Review by Woojin-Crive — APPROVED** [review]
LGTM
